src/components/Admin/Approval/Approval.js

```diff
@@ -78,71 +78,83 @@ class ImageApproval extends Component {
           <button className="button button-admin" onClick={this.back}>
             &laquo; Back
           </button>
         </div>
-        {this.state.images.map((image, index) => {
-          return (
-            <div key={image.id} className="preview-pane">
+        {image ? (
+          <div key={image.id} className="preview-pane">
+            <h2>
+              {this.state.storeList[image.label] !== undefined &&
+                this.state.storeList[image.label].name}
+            </h2>
+            <div>
               <img style={{maxHeight: 224}} src={image.url} alt="" />
-              <ItemSelector
-                item={image.label}
-                items={this.state.storeList}
-                setItem={cat => {
-                  this.changeCategory(image.id, cat);
-                  this.setState(prevState => {
-                    prevState.images[index].label = cat;
-                    return prevState;
-                  });
-                }}
-              />
-              <div>
-                <button
-                  className="button button-admin"
-                  data-id={image.id}
-                  data-index={index}
-                  data-label={image.label}
-                  onClick={this.trustImage}>
-                  Trust
-                </button>
-                <button
-                  className="button button-admin"
-                  data-id={image.id}
-                  data-index={index}
-                  onClick={this.setAsUnknown}>
-                  Unknown
-                </button>
-                <button
-                  className="button button-admin"
-                  data-id={image.id}
-                  data-index={index}
-                  data-label={image.label}
-                  onClick={this.deleteImage}>
-                  Delete
-                </button>
-              </div>
             </div>
-          );
-        })}
+            <ItemSelector
+              item={image.label}
+              items={Object.values(this.state.storeList)}
+              setItem={cat => {
+                this.changeCategory(image.id, cat);
+                this.setState(prevState => {
+                  prevState.image.label = cat;
+                  return prevState;
+                });
+              }}
+            />
+            <div>
+              <button
+                className="button button-admin"
+                data-id={image.id}
+                data-label={image.label}
+                onClick={this.trustImage}>
+                Trust
+              </button>
+              <button
+                className="button button-admin"
+                data-id={image.id}
+                onClick={this.setAsUnknown}>
+                Unknown
+              </button>
+              <button
+                className="button button-admin"
+                data-id={image.id}
+                data-label={image.label}
+                onClick={this.deleteImage}>
```

Comment:
Could these be individual anonymous functions? Means we can't pass by reference but the benefit is that we're not adding things to the dom just for this reason

```js
onClick={() => this.deleteImage(image)}

deleteImage(image) {
  // ...
}
```